### ISSUE TITLE
Recommendations for git/GitHub

### ISSUE THREAD

**Issue by milanmlft** [UCL-ARC/coding-standards#7]
Should we also add some recommendations regarding git/GitHub use?
E.g.

- [Commit messages](https://cbea.ms/git-commit/) (length, contents, ...)
- Branch names
- PR titles
- **PR merge strategy** (<-- I feel this one is most impactful)
- Branch protection rules
- Other GH settings like automatic branch deletion, required reviewers, ...
- ...

Also related: use of [semantic versioning](https://semver.org/)

**Comment by samcunliffe** [comment]
It's interesting that I think of most of these things as obvious. But yeah, maybe we should write this all down.

Branch names we de-facto always seem to use <initials>/<name of feature>
```
sc/fix-milans-terrible-production-critical-bug-issue-01
```
... just as a random example. Do we need to write down that we recommend <initials>/description-of-feature ?

**Comment by milanmlft** [comment]
I think that's not a terrible idea. I for one wasn't really used to that style but just started doing it as I saw that others were doing it in my projects.
[Another convention](https://dev.to/varbsan/a-simplified-convention-for-naming-branches-and-commits-in-git-il4) is to use `<category>/<description>`, where `<category>` is one of a limited set of `doc, fix, feat, test` for example.

**Comment by MonikaSvata** [comment]
If it doesn't exist, would it be useful to have a template repository - with the recommended settings?

**Comment by milanmlft** [comment]
> If it doesn't exist, would it be useful to have a template repository - with the recommended settings?

@paddyroddy might have some ideas with `safe-settings`?

**Comment by paddyroddy** [comment]
> [@paddyroddy](https://github.com/paddyroddy) might have some ideas with `safe-settings`?

Yeah, a lot of this stuff we can enforce with `github/safe-settings`. See my talk https://paddyroddy.github.io/talks/github-safe-settings-policy-as-code, and this PR https://github.com/UCL-ARC/.github/pull/24.
